### PR TITLE
Fix skeleton branch existing on setup

### PR DESCRIPTION
**Pull request by OCopping**

### CODE REVIEW

File: src/dls_python3_skeleton/__main__.py [hunk 1]
```diff
@@ -195,6 +227,10 @@ def main(args=None):
     sub.add_argument(
         "--package", default=None, help="Package name, defaults to directory name"
     )
+    # Add a command for cleaning an existing repo of skeleton code
+    sub = subparsers.add_parser("clean", help="Clean skeleton from existing repo")
```
Comment (coretl) [inline src/dls_python3_skeleton/__main__.py:231]:
```suggestion
    sub = subparsers.add_parser("clean", help="Clean up branch from failed skeleton merge")
```

**Review by coretl — CHANGES_REQUESTED** [review]

File: src/dls_python3_skeleton/__main__.py [hunk 2]
```diff
@@ -66,57 +66,69 @@ def replace_text(text: str) -> str:
...
+    branches = [x[2:] for x in str(git("branch", "--list")).split("\n")]
+
+    if f"{MERGE_BRANCH}" in branches:
+        raise Exception(
+            f"{MERGE_BRANCH} already exists. \
+                Please run 'dls-python3-skeleton clean' to remove it."
+        )
     else:
-        git("branch", "-d", MERGE_BRANCH, cwd=path)
+        with GitTemporaryDirectory() as git_tmp:
```
Comment (coretl) [inline src/dls_python3_skeleton/__main__.py:77]: You can ditch the `else:` and unindent this whole bock now you made the above an exception

File: tests/test_dls_python3_skeleton.py [hunk 3]
```diff
@@ -17,6 +18,12 @@ def check_output(*args, cwd=None) -> str:
         raise ValueError(e.output)
 
 
+def check_branches(module: str) -> List[str]:
+    return [
+        x[2:] for x in str(__main__.git("branch", "--list", cwd=module)).split("\n")
```
Comment (coretl) [inline tests/test_dls_python3_skeleton.py:23]: [Stack Overflow](https://stackoverflow.com/a/53271026) suggests using `git branch --format='%(refname:short)'`

File: tests/test_dls_python3_skeleton.py [hunk 4]
```diff
@@ -17,6 +18,15 @@ def check_output(*args, cwd=None) -> str:
         raise ValueError(e.output)
 
 
+def check_branches(module: str) -> List[str]:
+    return [
+        x
+        for x in __main__.git("branch", "--format=%(refname:short)", cwd=module).split(
+            "\n"
+        )
+    ]
```
Comment (coretl) [inline tests/test_dls_python3_skeleton.py:27]:
```suggestion
    return __main__.git("branch", "--format=%(refname:short)", cwd=module).split("\n")
```

File: src/dls_python3_skeleton/__main__.py [hunk 5]
```diff
@@ -66,6 +66,13 @@ def replace_text(text: str) -> str:
         text = text.replace("email@address.com", email)
         return text
 
+    branches = [x[2:] for x in str(git("branch", "--list", cwd=path)).split("\n")]
```
Comment (coretl) [inline src/dls_python3_skeleton/__main__.py:69]:
```suggestion
    branches = [x[2:] for x in str(git("branch", "--format=%(refname:short)", cwd=path)).split("\n")]
```

File: src/dls_python3_skeleton/__main__.py [hunk 6]
```diff
@@ -174,6 +184,20 @@ def existing(args):
     )
 
 
+def clean_repo(args):
+    path: Path = args.path
+
+    assert path.is_dir(), f"Expected {path} to be an existing directory"
+
+    branches = [x[2:] for x in str(git("branch", "--list")).split("\n")]
```
Comment (coretl) [inline src/dls_python3_skeleton/__main__.py:192]:
```suggestion
    branches = [x[2:] for x in str(git("branch", "--format=%(refname:short)", cwd=path)).split("\n")]
```
Although maybe this should be in a `def list_branches(path: Path) -> List[str]` function as it's used in 3 places?

Comment (OCopping) [reply src/dls_python3_skeleton/__main__.py:192]: Good point :thinking:

**Pull request merged**